src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -1255,17 +1255,16 @@ private BoundStatement ChainImplicitStructConstructor(MethodSymbol methodSymbol,
                 var codeGen = new CodeGen.CodeGenerator(method, block, builder, moduleBuilder, diagnosticsForThisMethod, optimizations, generateDebugInfo);
 
                 // We need to save additional debugging information for MoveNext of an async state machine.
-                var stateMachineMethod = method as SynthesizedStateMachineMethod;
-                bool isStateMachineMoveNextMethod = stateMachineMethod != null && method.Name == WellKnownMemberNames.MoveNextMethodName;
+                var stateMachineMoveNextMethod = method as SynthesizedStateMachineMoveNextMethod;
```

Comment:
In the C# code base we avoid type tests except to types that are internal APIs, which is usually evidenced by the fact that they introduce members that we need to use. The existing type test was to a type that did indeed have such members (e.g. `stateMachineMethod.StateMachineType`). That is why I suggested adding a virtual boolean property in `StateMachineMethod` for this purpose, though I think the previous version of the code that just used the Name property was appropriate.